### ISSUE TITLE
Turn exercise 012 into practice

### ISSUE THREAD

**Issue by cuihtlauac** [ocaml/ocaml.org#2180]
Extract the OCaml code from the markdown file corresponding to this exercise. The file is: `data/exercise/012_NAME.md` (where `012` is the number from the title of this issue and `NAME` is the rest of the file name, for instance `data/exercise/001_tail.md` is the file which corresponds to exercise `001`). Use the code in this file to create a folder `practice/012` (using this same number).

Populate the `012` folder with (almost) the same files as in folders `practice/001` and `practice/002` (they are examples).

Three files are different in each folder:
1. `practice/012/ex.ml`: it contains the function's signature, which needs to be implemented by the learner; 
1. `practice/012/work/impl.ml`: it includes the template meant to be filed by the learner;
1. `practice/012/answer/impl.ml`: consists of the solution from the original markdown file.

The examples included in the original markdown file must be turned into test cases in the `practice/012/ex.ml` file.

Folder structure to create:
```
practice/012/
├── answer
│   ├── impl.ml
│   └── test
│       ├── dune
│       └── run.ml
├── dune
├── dune-project
├── ex.ml
└── work
    ├── impl.ml
    └── test
        ├── dune

**Comment by maha-sachin** [comment]
@cuihtlauac @SaySayo , I am Maha from outreachy, Can you help me to assign this issue ?

**Comment by cuihtlauac** [comment]
Hi @maha-sachin, we assign issues once work has begun

**Comment by maha-sachin** [comment]
> Hi @maha-sachin, we assign issues once work has begun

@cuihtlauac , I'm interested in contributing to the project and I’d like to work on this.

**Comment by cuihtlauac** [comment]
@maha-sachin: You can start work on this

**Comment by maha-sachin** [comment]
> @maha-sachin: You can start work on this

@cuihtlauac , Thank you for the opportunity! I'll get started on it right away.

**Comment by cuihtlauac** [comment]
Fixed by #2218
